OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -1403,6 +1403,40 @@ public static ProcStatus IsInRange(DMProcState state) {
             return ProcStatus.Continue;
         }
 
+        public static ProcStatus AsType(DMProcState state) {
```

Comment:
A majority of this could be moved to a helper than both `IsType` and `AsType` use